clang/include/clang/Basic/Attr.td

```diff
@@ -1226,6 +1226,17 @@ def SYCLKernel : InheritableAttr {
   let Documentation = [SYCLKernelDocs];
 }
 
+def SYCLSpecialClass: InheritableAttr {
+  let Spellings = [Clang<"sycl_special_class">];
+  let Subjects = SubjectList<[CXXRecord]>;
+  let LangOpts = [SYCLIsDevice, SYCLIsHost];
+  let Documentation = [Undocumented];
+  let Args = [EnumArgument<"SpecialClassKind", "SpecialClassKind",
+                           ["accessor", "sampler", "stream", ""],
+                           ["Accessor", "Sampler", "Stream", "Generic"], 1>];
```

Comment:
If this is an internal-only attribute, should we give the user-facing enumeration names that are in the reserved namespace so that there's no potential for conflict with macros?

Also, if it's internal-only, do we really need a default argument or should we have folks be explicit with their intent?

Reply:
It is my understanding that  this is to be an internal-only attribute.

Reply:
Please take a look at https://github.com/intel/llvm/pull/2091/files#r454241718. If possible, we should be changing the definition here to be type agnostic and instead make the handling sensitive to presence of `__init` and `__finalize`.